.idea/MinStack.java

```diff
@@ -0,0 +1,42 @@
+/*
+Intuition : as this is requires min element - using PriorityQueue (min heap)
```

Comment:
Can you try doing it without using Priority queue , that would be optimal here and also mention complexity for individual functions separately in design problems